src/data/p5.TypedDict.js

```diff
@@ -0,0 +1,371 @@
+  if(this.data.hasOwnProperty(key)) {
+    delete this.data[key];
+    this.count--;
+  } else {
+    throw key + ' does not exist in this Dictionary';
+  }
+};
+
+
+/**
+ * Logs the list of items currently in the Dictionary
+ *
+ *
+ */
+p5.TypedDict.prototype.print = function() {
+  for (var item in this.data) {
+    console.log('key:' + item + ' value:' + this.data[item]);
+  }
+};
+
+p5.TypedDict.prototype._validate = function(key, value) {
+  return true;
+};
+
+// /**
+//  *
+//  * A simple Dictionary class for Strings.
+//  *
+//  *
+//  * @class p5.StringDict
+//  * @constructor
+//  * @extends p5.Element
+//  * @param {String}
+//  */
+p5.StringDict = function(key, value) {
+  this.data = {};
+  this.count = 0;
+
+  if(arguments.length === 1) {
+    if(key instanceof Object) {
+      this.data = key;
+      this.count = Object.keys(key).length;
+    }
+  } else {
+    this.data[key] = value;
+    this.count++;
+  }
+  return this;
+};
+
+
+p5.StringDict.prototype = Object.create(p5.TypedDict.prototype);
+
+
+p5.StringDict.prototype._validate = function(value) {
+  return (typeof value === 'string');
+};
+
+p5.StringDict.prototype.constructor = p5.StringDict;
+
+// /**
+//  *
+//  * A simple Dictionary class for Numbers.
+//  *
+//  *
+//  * @class p5.NumberDict
+//  * @constructor
+//  * @extends p5.TypedDict
+//  * @param
+//  */
+p5.NumberDict = function(key, value) {
+  this.data = {};
+  this.count = 0;
+
+  if(arguments.length === 1) {
+    if(key instanceof Object) {
+      this.data = key;
+      this.count = Object.keys(key).length;
+    }
+  } else {
+    this.data[key] = value;
+    this.count++;
+  }
+  return this;
+};
+
+p5.NumberDict.prototype = Object.create(p5.TypedDict.prototype);
+
+p5.NumberDict.prototype._validate = function(value) {
+  return (typeof value === 'number');
+};
+
+/**
+ * Add an amount to the current value in the Dictionary
+ *
+ *
+ */
+p5.NumberDict.prototype.add = function(key, amount) {
+  if(this.data.hasOwnProperty(key)){
+    this.data[key] += amount;
+  } else {
+    console.log('The key - ' + key + ' does not exist in this dictionary.');
+  }
+};
+
+/**
+ * Subtract an amount from a current value in the Dictionary
+ *
+ *
+ */
+p5.NumberDict.prototype.sub = function(key, amount) {
+  this.add(key, -amount);
+};
+
+/**
+ * Multiply a current value in the Dictionary
+ *
+ *
+ */
+p5.NumberDict.prototype.mult = function(key, amount) {
+  if(this.data.hasOwnProperty(key)){
+    this.data[key] *= amount;
+  } else {
+    console.log('The key - ' + key + ' does not exist in this dictionary.');
+  }
+};
+
+/**
+ * Divide a current value in the Dictionary
+ *
+ *
+ */
+p5.NumberDict.prototype.div = function(key, amount) {
+  if(this.data.hasOwnProperty(key)){
+    this.data[key] /= amount;
+  } else {
+    console.log('The key - ' + key + ' does not exist in this dictionary.');
+  }
+};
+
+/**
+ * Return the lowest key
+ *
+ *
+ */
+p5.NumberDict.prototype.minValue = function() {
```

Comment:
i have the feeling these four blocks could be reduced into one function, since only one symbol or line changes between them. for example, if it's just a < that flips.. you can't pass operators as arguments but you could write a function that takes 1 or -1, and multiplies both sides of the comparison by that operator. multiplying by -1 would have the effect of flipping the < to a >.